__tests__/outsideExecution.ts

```diff
@@ -0,0 +1,119 @@
+    },
+  ];
+
+  const options = {
+    caller: '0xabcdef',
+    nonce: '0x123',
+    execute_after: '100',
+    execute_before: '500',
+  };
+
+  it('should construct an OutsideExecution object', () => {
+    const oe = new OutsideExecution(calls, options);
+    expect(oe.calls).toEqual(calls);
+    expect(oe.options).toEqual(options);
+  });
+
+  it('should get typed data for V1', () => {
+    const oe = new OutsideExecution(calls, options);
+    const typedData = oe.getTypedData('SN_MAIN', EOutsideExecutionVersion.V1);
+
+    const expectedDomain = {
+      name: 'Account.execute_from_outside',
+      version: '1',
+      chainId: 'SN_MAIN',
+    };
+
+    const expectedTypes = OutsideExecutionTypesV1;
+
+    const expectedMessage = {
+      caller: options.caller,
+      nonce: options.nonce,
+      execute_after: options.execute_after,
+      execute_before: options.execute_before,
+      calls_len: calls.length.toString(),
+      calls: [
+        {
+          to: calls[0].contractAddress,
+          selector: BigInt(encodeShortString(calls[0].entrypoint)).toString(),
+          calldata_len: (calls[0].calldata ?? []).length.toString(),
+          calldata: calls[0].calldata ?? [],
+        },
+      ],
+    };
+
+    expect(typedData.domain).toEqual(expectedDomain);
+    expect(typedData.types).toEqual(expectedTypes);
+    expect(typedData.message).toEqual(expectedMessage);
+  });
+
+  it('should get typed data for V2', () => {
+    const oe = new OutsideExecution(calls, options);
+    const typedData = oe.getTypedData('SN_MAIN', EOutsideExecutionVersion.V2);
+
+    const expectedDomain = {
+      name: 'Account.execute_from_outside',
+      version: '2',
+      chainId: 'SN_MAIN',
+      revision: '1',
+    };
+
+    const expectedTypes = OutsideExecutionTypesV2;
+
+    const expectedMessage = {
+      Caller: options.caller,
+      Nonce: options.nonce,
+      'Execute After': options.execute_after,
+      'Execute Before': options.execute_before,
+      Calls: [
+        {
+          To: calls[0].contractAddress,
+          Selector: BigInt(encodeShortString(calls[0].entrypoint)).toString(),
+          Calldata: calls[0].calldata,
+        },
+      ],
+    };
+
+    expect(typedData.domain).toEqual(expectedDomain);
+    expect(typedData.types).toEqual(expectedTypes);
+    expect(typedData.message).toEqual(expectedMessage);
+  });
+
+  it('should get ABI data', () => {
+    const oe = new OutsideExecution(calls, options);
+    const abiData = oe.getABIData();
+
+    const expectedResult = {
+      caller: options.caller,
+      nonce: options.nonce,
+      execute_after: options.execute_after,
+      execute_before: options.execute_before,
+      calls: [
+        {
+          to: calls[0].contractAddress,
+          selector: BigInt(encodeShortString(calls[0].entrypoint)).toString(),
+          calldata: calls[0].calldata,
+        },
+      ],
+    };
+
+    expect(abiData).toEqual(expectedResult);
+  });
+});
```

Comment:
Is there an Acc implementing this snip so we can test it on Account?
If so I would like to have a full e2e test with ACC using it.

Reply:
Yep, Argent and Braavos support it at least on Sepolia, I'll write the missing test in a few days, sorry for the delay

Reply:
I just realized that OZ account contract doesn't support SNIP-9 yet. And both Argent and Braavos are distributed under GPL, so I can't just take their compiled contracts and put them into `__mocks__` directory. I can request a feature on OZ but nobody knows how long it will take. If I compile a "dummy" account contract with minimal `execute_from_outside` implementation and use it in tests, will it count as a valid test?

Reply:
Tagging Argent folks, @janek26 @dhruvkelawala hi! for testing purposes, does GPL license allow usage of your Account contract? @kfastov I don't think we would have problems with the GPL license, this is a non-profit library in any case :)

Reply:
@ivpavici all good to use our contracts. double checked it with Sergio.